src/main/java/seedu/address/logic/commands/schedule/RescheduleCommand.java

```diff
@@ -0,0 +1,179 @@
+package seedu.address.logic.commands.schedule;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.schedule.CliSyntax.PREFIX_SESSION_INDEX;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_SCHEDULES;
+
+import java.util.List;
+import java.util.Optional;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.client.Client;
+import seedu.address.model.schedule.Schedule;
+import seedu.address.model.session.Session;
+
+/**
+ * Edits the details of an existing Client in the address book.
+ */
+public class RescheduleCommand extends Command {
+
+    public static final String COMMAND_WORD = "reschedule";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Reschedule a client with another session. "
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + PREFIX_SESSION_INDEX + "SESSION "
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_SESSION_INDEX + "1 ";
+
+    public static final String MESSAGE_EDIT_SCHEDULE_SUCCESS = "Rescheduled : \n%1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_SCHEDULE = "This Schedule overlaps with an existing Schedule";
+
+    private final Index index;
+    private final Index sessionIndex;
+    private final RescheduleDescriptor editRescheduleDescriptor;
+
+    /**
+     * @param index of the Schedule in the filtered schedule list to edit
+     * @param editRescheduleDescriptor details to edit the schedule with
+     */
+    public RescheduleCommand(Index index, Index sessionIndex, RescheduleDescriptor editRescheduleDescriptor) {
```

Comment:
should the variable name index be changed? since there are 2 "index"

Reply:
okay sure I'll change that